### ISSUE TITLE
MobileFormatter - DateTime.Kind lost after deserialization

### ISSUE THREAD

**Issue by rockfordlhotka** [MarimerLLC/csla#3363]
### Discussed in https://github.com/MarimerLLC/csla/discussions/3362

<div type='discussions-op-text'>

<sup>Originally posted by **GlenYappa** June  2, 2023</sup>
I am in the process of upgrading a large business object library from Csla 4 to Csla 6.2.2 while at the same time moving from .Net Framework 4.7.2 to .Net Standard 2.0. One issue that I am having is DateTime properties are losing the DateTime.Kind when deserialized. In the example below I have a simple ReadOnly csla object with a nullable Datetime and I am serializing/deserializing with the MobileFormatter directly just for test purposes. You can see that after deserialization the Kind value is Unspecified. Is this a known issue? 

	var t = TestHarness.App.DataPortalFactory.GetPortal<TestHarness.DateTimeTest>().Create();
	t.ID = Guid.NewGuid();
	t.TestDate = DateTime.Now;

	Console.WriteLine(t);

	var formatter = new Csla.Serialization.Mobile.MobileFormatter(TestHarness.App.AppContext);

	byte[] bytes = new byte[0];
	using (MemoryStream stream = new MemoryStream())
	{
		formatter.Serialize(stream, t);
		bytes = stream.ToArray();
	}

	using (MemoryStream stream = new MemoryStream(bytes))
	{
		t = (TestHarness.DateTimeTest)formatter.Deserialize(stream);
	}

	Console.WriteLine(t);

ID = ba5ddc50-75ea-4b7d-b656-7fc12faab73e
TestDate = 6/2/2023 8:59:27 AM
DateTime Kind = Local

ID = ba5ddc50-75ea-4b7d-b656-7fc12faab73e
TestDate = 6/2/2023 8:59:27 AM
DateTime Kind = Unspecified

Thanks</div>

**Comment by rockfordlhotka** [comment]
Maybe can be resolved once #2531 is complete?

**Comment by rockfordlhotka** [comment]
I can't replicate this in CSLA 9. I added this test:

```c#
    [TestMethod]
    public void DateTimeKind()
    {
      var portal = _testDIContext.CreateDataPortal<DateTimeHolder>();
      var obj = portal.Create();
      DateTime.SpecifyKind(obj.Value, System.DateTimeKind.Local);
      var obj2 = obj.Clone();
      Assert.AreEqual(obj.Value.Kind, obj2.Value.Kind);
    }
```

And it passes. I tried with the original kind as Utc as well.
